lib/widgets/message_list.dart

```diff
@@ -1347,40 +1370,61 @@ class MessageWithPossibleSender extends StatelessWidget {
 
     return GestureDetector(
       behavior: HitTestBehavior.translucent,
-      onLongPress: () => showMessageActionSheet(context: context, message: message),
+      onLongPress: () async {
+        statesController.update(WidgetState.selected, true);
+        if (context.mounted) {
+          ModalStatus status = showMessageActionSheet(
+              context: context,
+              message: message);
+          await status.closed;
+          statesController.update(WidgetState.selected, false);
+        }
+      },
+      onLongPressDown: (_) => statesController.update(WidgetState.pressed, true),
+      onLongPressCancel: () => statesController.update(WidgetState.pressed, false),
+      onLongPressUp: () => statesController.update(WidgetState.pressed, false),
       child: Padding(
         padding: const EdgeInsets.symmetric(vertical: 4),
-        child: Column(children: [
-          if (senderRow != null)
-            Padding(padding: const EdgeInsets.fromLTRB(16, 2, 16, 0),
-              child: senderRow),
-          Row(
-            crossAxisAlignment: CrossAxisAlignment.baseline,
-            textBaseline: localizedTextBaseline(context),
-            children: [
-              const SizedBox(width: 16),
-              Expanded(child: Column(
-                crossAxisAlignment: CrossAxisAlignment.stretch,
-                children: [
-                  MessageContent(message: message, content: item.content),
-                  if ((message.reactions?.total ?? 0) > 0)
-                    ReactionChipsList(messageId: message.id, reactions: message.reactions!),
-                  if (editStateText != null)
-                    Text(editStateText,
-                      textAlign: TextAlign.end,
-                      style: TextStyle(
-                        color: designVariables.labelEdited,
-                        fontSize: 12,
-                        height: (12 / 12),
-                        letterSpacing: proportionalLetterSpacing(
-                          context, 0.05, baseFontSize: 12))),
-                ])),
-              SizedBox(width: 16,
-                child: message.flags.contains(MessageFlag.starred)
-                  ? Icon(ZulipIcons.star_filled, size: 16, color: designVariables.star)
-                  : null),
-            ]),
-        ])));
+        child: DecoratedBox(
```

Comment:
Let's actually swap this `DecoratedBox` with its parent `Padding`. That way, the tint also applies on the padding at the top and bottom, like in the [Figma](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3988-34789&m=dev).